src/foam/nanos/session/Session.js

```diff
@@ -243,6 +247,59 @@ foam.CLASS({
           .put("group", group)
           .put("appConfig", group.getAppConfig(rtn));
       `
+    },
+    {
+      name: 'validate',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaCode: `
+        if ( getUserId() < 0 ) {
+          throw new IllegalStateException("User id is invalid.");
+        }
+
+        if ( getAgentId() < 0 ) {
+          throw new IllegalStateException("Agent id is invalid.");
+        }
+
+        if ( getUserId() > 0 ) {
+          checkUserEnabled(x, getUserId());
+        }
+
+        if ( getAgentId() > 0 ) {
+          checkUserEnabled(x, getAgentId());
+
+          DAO agentJunctionDAO = (DAO) x.get("agentJunctionDAO");
+          UserUserJunction junction = (UserUserJunction) agentJunctionDAO.find(
+            AND(
+              EQ(UserUserJunction.SOURCE_ID, getAgentId()),
+              EQ(UserUserJunction.TARGET_ID, getUserId())
+            )
+          );
+
+          if ( junction == null ) {
+            throw new RuntimeException("The junction between user and agent was not found.");
+          }
+        }
+      `
+    },
+    {
+      name: 'checkUserEnabled',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'userId', type: 'Long' }
+      ],
+      javaCode: `
+        User user = (User) ((DAO) x.get("localUserDAO")).inX(x).find(userId);
```

Comment:
I think it makes more sense to use the system context here because we want to make sure we will be allowed to find the user.

```suggestion
        User user = (User) ((DAO) x.get("localUserDAO")).find(userId);
```

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/2814/commits/af11f002f478b55b6dc63691f4b907f00858349f.